File: backend/grafioschtrader-common/src/main/java/grafioschtrader/error/SingleNativeMsgError.java
package grafioschtrader.error;

/**
 * Most single message error should transformed in this class for a response.
 *
 * @author Hugo Graf
 *
 */
public class SingleNativeMsgError {
  public String message;

  public SingleNativeMsgError(String message) {
    this.message = message;
  }

  public SingleNativeMsgError(String message, boolean onlyFirstLine) {
    this.message = message.split("\r\n|\r|\n")[0];
  }
}

File: frontend/src/app/shared/login/service/single.native.msg.error.ts
import {GetTransformedError} from './get.transformed.error';
import {TransformedError} from './transformed.error';

export class SingleNativeMsgError implements GetTransformedError {
  public message: string;

  getTransformedError(): TransformedError {
    const transformedError = new TransformedError();
    transformedError.errorClass = this;
    transformedError.msg = this.message;
    return transformedError;
  }
}
